### PR TITLE
[test] Update chip_sw_pin_mux testpoint

### LINKED ISSUE #14146: [chip-test] chip_sw_pin_mux
### Test point name

[chip_sw_pin_mux](https://github.com/lowRISC/opentitan/blob/b65e2705eb4ee814b46939d44f84d1adfee133c3/hw/top_earlgrey/data/chip_testplan.hjson#L351)

### Host side component

SystemVerilog

### OpenTitanTool infrastructure implemented

_No response_

### Contact person

@msfschaffner

### Checklist

Please fill out this checklist as items are completed. Link to PRs and issues as appropriate.
- [x] Check if existing test covers most or all of this testpoint (if so, either extend said test to cover all points, or skip the next 3 checkboxes)
- [ ] Device-side (C) component developed
- [ ] Bazel build rules developed
- [x] Host-side component developed
- [x] HJSON test plan updated with test name (so it shows up in the dashboard)
- [x] Test added to dvsim nightly regression (and passing at time of checking)

### PR DESCRIPTION
**Pull request by msfschaffner**
This fixes #14146.

The remaining TODOs are tracked in separate issues.

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

**Comment by msfschaffner** [comment]
CC @sha-ron @ariel9678

**Review by weicaiyang — APPROVED** [review]
should we also add the chip_tap_straps_* test, which proves that we rv_dm tap, LC tap and DFT tap via pinmux IOs?
Otherwise, LGTM

**Comment by msfschaffner** [comment]
> should we also add the chip_tap_straps_* test, which proves that we rv_dm tap, LC tap and DFT tap via pinmux IOs? Otherwise, LGTM

I am going to hold off merging this since @sriyerg mentioned that his padctrl test could actually cover this testpoint entirely so that we would not have to cross reference all these tests.

**Comment by tjaychen** [comment]
wait for things like attaching pwm / i2c to a particular pin, wouldn't we still need the xref tests..? I thought the `padctrl` would be more like a gpio test.

**Comment by msfschaffner** [comment]
I am holding off until @sriyerg has merged #15814 which updates the description of the `chip_sw_pin_mux` testpoint.
Once that has landed, I will check again and if necessary make the xrefs.

**Comment by tjaychen** [comment]
ah sounds good.

File: hw/top_earlgrey/data/chip_testplan.hjson
Comment (sriyerg) [inline hw/top_earlgrey/data/chip_testplan.hjson:411]: These IO tests do exercise the muxing logic to connect the right signals from the peripherals to the right pads, but I am not sure if these cross-references are needed anymore, given the test in my PR fully stresses the mux (which is the point of this test).

**Comment by msfschaffner** [comment]
Given that this is now superseded by https://github.com/lowRISC/opentitan/pull/15814, are we ok with closing this down without any further Xrefs in the testplan?

**Comment by tjaychen** [comment]
yep! go for it!

**Comment by tjaychen** [comment]
oops! accidentally closed. Sorry about that.

**Comment by msfschaffner** [comment]
No worries :) let's close.

**Pull request closed without merging**
